### PR TITLE
Redux sagas

### PR DESCRIPTION
**Pull request by stevennx**
image

### CODE REVIEW

File: src/components/login/login.component.js [hunk 1]
```diff
@@ -1,43 +1,48 @@
...
+                    <button type="submit" className="btn btn-primary mr-2">Login</button>
+                    <button type="button" className="btn btn-secondary">Signup</button>
+                </div>
+            </Form>
+        )} />
 );
 
-
-export default Login;
+const mapDispatchToProps = dispatch => (bindActionCreators({loginUserAction}, dispatch))
```
Comment (gzcharleszhang) [inline src/components/login/login.component.js]: i think this should be in a container

File: src/components/login/login.services.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+
+// export default request => {
```
Comment (gzcharleszhang) [inline src/components/login/login.services.js]: if every service have one of these, that's a lot of duplicated code, maybe we can abstract away some of the code?

File: src/components/login/login.component.js [hunk 3]
```diff
@@ -1,43 +1,48 @@
...
+        .email("Invalid email"),
     password: Yup.string()
-    .min(5, "Too Short!")
-    .max(50, "Too Long!")
+        .min(5, "Too Short!")
+        .max(50, "Too Long!")
 });
 
-const Login = () => (
+const LoginPage = (props) => (
```
Comment (gzcharleszhang) [inline src/components/login/login.component.js]: can you use object destructuring here for `prop`? Also, we will need to use `PropTypes` for every component that has props

File: src/components/login/login.reducers.js [hunk 4]
```diff
@@ -7,9 +7,11 @@ const initialState = {
 
 export default (state = initialState, action) => {
     switch (action.type) {
-      case LOGIN_USER:
-        return {...state, user: get(action.payload, undefined)}
-      default:
-        return state;
+        case LOGIN_USER_SUCCESS:
+            return { ...state, user: get(action.payload, undefined) }
```
Comment (gzcharleszhang) [inline src/components/login/login.reducers.js]: I think null is a better value than undefined

File: src/components/login/login.reducers.js [hunk 5]
```diff
@@ -7,9 +7,11 @@ const initialState = {
...
+        case LOGIN_USER_ERROR:
+            return { ...state, user: get(action.payload, undefined) }
```
Comment (gzcharleszhang) [inline src/components/login/login.reducers.js]: same as above

File: src/components/login/login.services.js (hunk 2 above)
Comment (stevennx) [reply src/components/login/login.services.js]: Yeah we could have a common service

File: src/components/login/login.component.js (hunk 1 above)
Comment (stevennx) [reply src/components/login/login.component.js]: So any wrapper that produces a higher-level component should go in container.js? 
For ex. using withFormik(LoginPage)

Comment (gzcharleszhang) [reply src/components/login/login.component.js]: anything that involves redux or logic i.e. props, actions, handlers

Comment (gzcharleszhang) [reply src/components/login/login.component.js]: ideally yeah, Formik stuff should be in a container since it's logic

**Pull request merged**